### PR TITLE
Combobox filtering behavior

### PR DESCRIPTION
**Pull request by udimberto**
### What does it do?
Updates the filtering rule for the `Combobox` component.

Before, just showing options which starts with the typed term.
Now, displays options containing any part of the typed term.

### Why is it needed?
Will help out to search for User Repositories at the Strapi Cloud Dashboard.

### How to test it?
Unit tests.
E2E existent tests.

### CODE REVIEW

File: packages/strapi-design-system/src/Combobox/__tests__/utils.spec.js [hunk 1]
```diff
@@ -6,9 +6,48 @@ describe('Combobox | utils', () => {
...
+      });
+
+      it('2 last of 5 items', () => {
+        const result = filterOptions(options, 'strapi');
+        const expected = [options[3], options[4]];
+
+        expect(result).toEqual(expected);
+      });
+
+      it('1 item', () => {
```
Comment (gu-stav) [inline packages/strapi-design-system/src/Combobox/__tests__/utils.spec.js:43]: What do you think about adding one more test, to prove the edge-case `filterOptions(options, '')` or `filterOptions(options, null)`is working well?

File: packages/strapi-design-system/src/Combobox/utils.js [hunk 2]
```diff
@@ -30,10 +30,13 @@ export const TreeActions = {
 // filter an array of options against an input string
 // returns an array of options that begin with the filter string, case-independent
 export function filterOptions(options = [], filter, exclude = []) {
-  return filter
+  const equalizedTerm = String(filter || '').toLowerCase();
```
Comment (gu-stav) [inline packages/strapi-design-system/src/Combobox/utils.js:33]: What do you think about using:

```suggestion
  const equalizedTerm = String(filter ?? '').toLowerCase();
```

instead?

```js
String(0 || '') // ''
String(0 ?? '') // '0'
```

File: packages/strapi-design-system/src/Combobox/__tests__/utils.spec.js [hunk 3]
```diff
@@ -6,9 +6,48 @@ describe('Combobox | utils', () => {
...
       const expected = [{ props: { children: 'Hamburger' } }];
+
       expect(actual).toEqual(expected);
     });
 
+    describe('should filter the options correctly, with partial term', () => {
+      const options = [
+        { props: { children: 'hicham/project-super-cms' } },
+        { props: { children: 'pierre/project-awesome-cms' } },
+        { props: { children: 'udi/cool-cms' } },
```
Comment (HichamELBSI) [inline packages/strapi-design-system/src/Combobox/__tests__/utils.spec.js:17]: Not sure that we should keep the Strapier names in the codebase 😅. Can you rename these values please 👍🏽

File: packages/strapi-design-system/src/Combobox/__tests__/utils.spec.js (hunk 1 above)
Comment (HichamELBSI) [reply packages/strapi-design-system/src/Combobox/__tests__/utils.spec.js:43]: +1

**Review by HichamELBSI — CHANGES_REQUESTED** [review]

File: packages/strapi-design-system/src/Combobox/__tests__/utils.spec.js (hunk 1 above)
Comment (udimberto) [reply packages/strapi-design-system/src/Combobox/__tests__/utils.spec.js:86]: Cool.
Those test-cases already exists.
I've re-order them now, adding one more.
Thank you

File: packages/strapi-design-system/src/Combobox/utils.js (hunk 2 above)
Comment (udimberto) [reply packages/strapi-design-system/src/Combobox/utils.js]: Awesome!
Thank you.

File: packages/strapi-design-system/src/Combobox/__tests__/utils.spec.js (hunk 3 above)
Comment (udimberto) [reply packages/strapi-design-system/src/Combobox/__tests__/utils.spec.js]: Sorry about that.
Fixed.
Thank you.

**Review by HichamELBSI — APPROVED** [review]

**Review by gu-stav — APPROVED** [review]
Well done! :)
